Ghidra/Processors/SuperH/data/languages/superh.sinc

```diff
@@ -0,0 +1,1410 @@
+{
+    *:4 (rn_08_11 + disp) = rm_04_07;
+}
+
+:mov.b  @(r0,rm_04_07),rn_08_11  is r0 & opcode_12_15=0b0000 & rn_08_11 & rm_04_07 & opcode_00_03=0b1100 
+{
+    rn_08_11 = sext(*:1 (rm_04_07 + r0));
+}
+
+:mov.w  @(r0,rm_04_07),rn_08_11  is r0 & opcode_12_15=0b0000 & rn_08_11 & rm_04_07 & opcode_00_03=0b1101 
+{
+    rn_08_11 = sext(*:2 (rm_04_07 + r0));
+}
+
+:mov.l  @(r0,rm_04_07),rn_08_11  is r0 & opcode_12_15=0b0000 & rn_08_11 & rm_04_07 & opcode_00_03=0b1110 
+{
+    rn_08_11 = *:4 (rm_04_07 + r0);
+}
+
+:mov.b  rm_04_07,@(r0,rn_08_11)  is r0 & opcode_12_15=0b0000 & rn_08_11 & rm_04_07 & opcode_00_03=0b0100 
+{
+    *:1 (rn_08_11 + r0) = rm_04_07;
+}
+
+:mov.w  rm_04_07,@(r0,rn_08_11)  is r0 & opcode_12_15=0b0000 & rn_08_11 & rm_04_07 & opcode_00_03=0b0101 
+{
+    *:2 (rn_08_11 + r0) = rm_04_07;
+}
+
+:mov.l  rm_04_07,@(r0,rn_08_11)  is r0 & opcode_12_15=0b0000 & rn_08_11 & rm_04_07 & opcode_00_03=0b0110 
+{
+    *:4 (rn_08_11 + r0) = rm_04_07;
+}
+
+:mov.b  @(disp_00_07,gbr),r0  is gbr & r0 & opcode_08_15=0b11000100 & disp_00_07  
+{
+    r0 = sext(*:1 (gbr + disp_00_07));
+}
+
+:mov.w  @(disp,gbr),r0  is gbr & r0 & opcode_08_15=0b11000101 & disp_00_07  [disp = (disp_00_07 << 1); ]
+{
+    r0 = sext(*:2 (gbr + disp));
+}
+
+:mov.l  @(disp,gbr),r0  is gbr & r0 & opcode_08_15=0b11000110 & disp_00_07  [disp = (disp_00_07 << 2); ]
+{
+    r0 = *:4 (gbr + disp);
+}
+
+:mov.b  r0,@(disp_00_07,gbr)  is r0 & gbr & opcode_08_15=0b11000000 & disp_00_07  
+{
+    *:1 (gbr + disp_00_07) = r0;
+}
+
+:mov.w  r0,@(disp,gbr)  is r0 & gbr & opcode_08_15=0b11000001 & disp_00_07 [disp = (disp_00_07 << 1); ]
+{
+    *:2 (gbr + disp) = r0;
+}
+
+:mov.l  r0,@(disp,gbr)  is r0 & gbr & opcode_08_15=0b11000010 & disp_00_07 [disp = (disp_00_07 << 2); ]
+{
+    *:4 (gbr + disp) = r0;
+}
+
+:movt   rn_08_11  is opcode_12_15=0b0000 & rn_08_11 & opcode_00_07=0b00101001 
+{
+    rn_08_11 = zext($(T_FLAG));
+}
+
+:swap.b rm_04_07,rn_08_11  is opcode_12_15=0b0110 & rn_08_11 & rm_04_07 & opcode_00_03=0b1000 
+{
+    local temp0;
+    local temp1;
+    
+    temp0 = rm_04_07 & 0xFFFF0000;
+    temp1 = (rm_04_07 & 0x000000FF) << 8;
+    
+    rn_08_11 = (rm_04_07 & 0x0000FF00) >> 8;
+    rn_08_11 = rn_08_11 | temp1 | temp0;
+}
+
+:swap.w rm_04_07,rn_08_11  is opcode_12_15=0b0110 & rn_08_11 & rm_04_07 & opcode_00_03=0b1001 
+{
+    local temp;
+    
+    temp = (rm_04_07 >> 16) & 0x0000FFFF;
+    
+    rn_08_11 = rm_04_07 << 16;
+    rn_08_11 = rn_08_11 | temp;
+}
+
+:xtrct  rm_04_07,rn_08_11  is opcode_12_15=0b0010 & rn_08_11 & rm_04_07 & opcode_00_03=0b1101 
+{
+    local high;
+    local low;
+    
+    high = (rm_04_07 << 16) & 0xFFFF0000;
+    low = (rn_08_11 >> 16) & 0x0000FFFF;
+    rn_08_11 = high | low;
+}
+
+#
+# Arithmetic Operation Instructions
+#
+:add    rm_04_07,rn_08_11  is opcode_12_15=0b0011 & rn_08_11 & rm_04_07 & opcode_00_03=0b1100 
+{
+    rn_08_11 = rn_08_11 + rm_04_07;
+}
+
+:add    imm_00_07,rn_08_11  is opcode_12_15=0b0111 & rn_08_11 & imm_00_07  
```

Comment:
You should include a sub-constructor for immediate values which will do the sign extending for you and the formatting. Something like:
`immed8: "#"^imm_00_07 is imm_00_07 {temp:4 = imm_00_07; export temp;}`

Compare the difference with add and other instructions with a number like -1.